db.go

```diff
@@ -322,8 +428,17 @@ func Open(dir string, l log.Logger, r prometheus.Registerer, opts *Options) (db
 		minValidTime = blocks[len(blocks)-1].Meta().MaxTime
 	}
 
-	if err := db.head.Init(minValidTime); err != nil {
-		return nil, errors.Wrap(err, "read WAL")
+	if initErr := db.head.Init(minValidTime); initErr != nil {
+		err := errors.Cause(initErr) // So that we can pick up errors even when wrapped.
+		if _, ok := err.(*wal.CorruptionErr); ok {
```

Comment:
Can we have separate PR for this?

Reply:
No there is no error, this should be part of this PR. 
`db.head.Init` was doing a repair when the WAL is corrupted and with this change this repair is now moved in tsdb.Open. 
This is needed so that when we create a querier need to load the WAL for the Head but not try to repair it when corrupted.

If you look at the refactored test it should make it more clear  why this change was needed.

Reply:
OK, can agree, will rereview